pybamm/models/base_model.py

```diff
@@ -421,83 +421,176 @@ def input_parameters(self):
             self._input_parameters = self._find_symbols(pybamm.InputParameter)
         return self._input_parameters
 
-    def get_parameter_info(self):
+    def get_parameter_info(self, by_submodel=False):
         """
         Extracts the parameter information and returns it as a dictionary.
         To get a list of all parameter-like objects without extra information,
         use :py:attr:`model.parameters`.
+
+        Parameters
+        ----------
+        by_submodel : bool, optional
+            Whether to return the parameter info sub-model wise or not (default False)
         """
         parameter_info = {}
-        parameters = self._find_symbols(pybamm.Parameter)
-        for param in parameters:
-            parameter_info[param.name] = (param, "Parameter")
-
-        input_parameters = self._find_symbols(pybamm.InputParameter)
-        for input_param in input_parameters:
-            if not input_param.domain:
-                parameter_info[input_param.name] = (input_param, "InputParameter")
-            else:
-                parameter_info[input_param.name] = (
-                    input_param,
-                    f"InputParameter in {input_param.domain}",
-                )
 
-        function_parameters = self._find_symbols(pybamm.FunctionParameter)
-        for func_param in function_parameters:
-            if func_param.name not in parameter_info:
-                input_names = "', '".join(func_param.input_names)
-                parameter_info[func_param.name] = (
-                    func_param,
-                    f"FunctionParameter with inputs(s) '{input_names}'",
-                )
+        if by_submodel:
```

Comment:
right now you are only checking the variables for submodels - you also need to check `rhs`, `algebraic`, etc. as in `_find_symbols`. I think it makes sense to make it so that `_find_symbols` can either act on the whole model or by submodel, or to make a `_find_symbols_by_sumbodel` method that will look something like 

```python3

    def _find_symbols(self, typ, submodel):
        """Find all the instances of `typ` in the model"""
        unpacker = pybamm.SymbolUnpacker(typ)
        all_input_parameters = unpacker.unpack_list_of_symbols(
            list(self.submodels[submodel].rhs.values())
            + list(self.submodels[submodel].algebraic.values())
            + list(self.submodels[submodel].initial_conditions.values())
            + [
                x[side][0]
                for x in self.submodels[submodel].boundary_conditions.values()
                for side in x.keys()
            ]
            + list(self.variables_by_submodel[submodel].values())
            + [event.expression for event in self.submodels[submodel].events]
        )
        return list(all_input_parameters)
```

Reply:
Thanks for pointing it out! I was about to ask this sooner.